crates/tap-agent/src/agent/sender_account.rs

```diff
@@ -129,56 +203,122 @@ pub enum SenderAccountMessage {
 pub struct SenderAccount;
 
+/// Arguments received in startup while spawing [SenderAccount] actor
 pub struct SenderAccountArgs {
     pub config: &'static SenderAccountConfig,
 
+    /// Connection to database
     pub pgpool: PgPool,
+    /// Current sender address
     pub sender_id: Address,
+    /// Watcher that returns a list of escrow accounts for current indexer
     pub escrow_accounts: Receiver<EscrowAccounts>,
-    pub indexer_allocations: Receiver<HashSet<Address>>,
+    /// Watcher that returns a set of open and recently closed allocation ids
+    pub indexer_allocations: Receiver<HashSet<AllocationId>>,
+    /// SubgraphClient of the escrow subgraph
     pub escrow_subgraph: &'static SubgraphClient,
+    /// SubgraphClient of the network subgraph
     pub network_subgraph: &'static SubgraphClient,
+    /// Domain separator used for tap
     pub domain_separator: Eip712Domain,
+    /// Endpoint URL for aggregator server
     pub sender_aggregator_endpoint: Url,
-    pub allocation_ids: HashSet<Address>,
+    /// List of allocation ids that must created at startup
+    pub allocation_ids: HashSet<AllocationId>,
+    /// Prefix used to bypass limitations of global actor registry (used for tests)
     pub prefix: Option<String>,
 
+    /// Configuration for retry scheduler in case sender is denied
     pub retry_interval: Duration,
 }
+
+/// State used by the actor
+///
+/// This is a separate instance that makes it easier to have mutable
+/// reference, for more information check ractor library
 pub struct State {
+    /// Prefix used to bypass limitations of global actor registry (used for tests)
     prefix: Option<String>,
+    /// Tracker used to monitor all pending fees across allocations
+    ///
+    /// Since rav requests are per allocation, this also has the algorithm
+    /// to select the next allocation to have a rav request.
+    ///
+    /// This monitors if rav requests succeeds or fails and apply proper backoff.
+    ///
+    /// Keeps track of the buffer returning values for both inside or outside the buffer.
+    ///
+    /// It selects the allocation with most amount of pending fees.
+    /// Filters out allocations in the algorithm in case:
+    ///     - In back-off
+    ///     - Marked as closing allocation (blocked)
+    ///     - Rav request in flight (selected the previous time)
     sender_fee_tracker: SenderFeeTracker,
+    /// Simple tracker used to monitor all Ravs that were not redeemed yet.
+    ///
+    /// This is used to monitor both active allocations and closed but not redeemed.
     rav_tracker: SimpleFeeTracker,
+    /// Simple tracker used to monitor all invalid receipts ever.
     invalid_receipts_tracker: SimpleFeeTracker,
-    allocation_ids: HashSet<Address>,
+    /// Set containing current active allocations
+    allocation_ids: HashSet<AllocationId>,
+    /// Keeps a reference of the handle for indexer_allocation pipe
     _indexer_allocations_handle: JoinHandle<()>,
```

Comment:
@gusinacio could we add a quick comment on why we're underscoring these parameters please 🙏 ?

Reply:
I just removedall JoinHandles. It was created with expectations that the Drop trait would cancel the task. But I just checked the docs and it's not the case.